connectors/sources/mysql.py

```diff
@@ -218,11 +216,21 @@ async def ping(self):
             if not self.ssl_disabled
             else None,
         }
+
+        if self._connection_pool is None:
+            self._connection_pool = await aiomysql.create_pool(**connection_string)
+
+        return self._connection_pool
```

Comment:
As we are storing in instance variable `_connection_pool` then we can avoid returning and can be directly used `self.__connection_pool`